corehq/apps/translations/integrations/transifex/client.py

```diff
@@ -17,42 +18,76 @@
 
 class TransifexApiClient(object):
     def __init__(self, token, organization, project, use_version_postfix=True):
-        self.username = API_USER
-        self.token = token
-        self.organization = organization
-        self.project = project
         self.use_version_postfix = use_version_postfix
+        self.api = TransifexApi(auth=token)
+        self.organization = self._get_organization(organization)
+        self.project = self._get_project(project)
 
     @property
-    def _auth(self):
-        return self.username, self.token
+    def _i18n_format(self):
+        return self.api.I18nFormat(id="PO")
 
     def _create_resource(self, resource_slug, resource_name):
-        ...
+        return self.api.Resource.create(
+            name=resource_name,
+            slug=resource_slug,
+            project=self.project,
+            i18n_format=self._i18n_format
+        )
+
+    @staticmethod
+    def _upload_content(cls, content, **kwargs):
```

Comment:
This is primarily an effort at DRYness, rather than duplicating this whole method including its error handling, which would be done inside both `_upload_resource_strings` and `_upload_resource_translations`.

Reply:
Okay, Good to still consider the renaming though.